### ISSUE TITLE
window.close should close opened connections

### ISSUE THREAD

**Issue by nicolashenry** [tmpvar/jsdom#1197]
window.close should abort opened connections like resource requests and XMLHTTPRequest like the browser do on tab closing.

My use case : 
```javascript
// I do some (long) script and ajax calls
...
// I want to stop everything if the whole process take more than 30 seconds 
setTimeout(function () {
	next(new Error('timeout'));
	window.close();
}, 30 * 1000);
```
I get errors after that from the request callbacks because the window.document does not exist anymore.

**Comment by domenic** [comment]
PR welcome!

**Comment by domenic** [comment]
Fixed by 2c8933905e9b81436843fe8d429918876a277e54!
